ci/jenkins/test-flow-visibility.sh

```diff
@@ -0,0 +1,145 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+FAILURE=false
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+GIT_CHECKOUT_DIR=${THIS_DIR}/..
+
+_usage="Usage: $0 [--kubeconfig <KubeconfigSavePath>]
+
+Setup Grafana, ClickHouse and Antrea Flow Aggregator to validate flow-aggregator.yml and flow-visibility.yml.
+
+    --kubeconfig    Path to kubeconfig.
+
+"
+
+echoerr() {
+  >&2 echo "$@"
+}
+
+print_usage() {
+  echoerr "$_usage"
+}
+
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+case $key in
+    --kubeconfig)
+    export KUBECONFIG="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+start_antrea() {
+  echo "=== Starting Antrea ==="
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  kubectl rollout status --timeout=5m deployment/coredns -n kube-system
+  kubectl rollout status --timeout=5m deployment.apps/antrea-controller -n kube-system
+  kubectl rollout status --timeout=5m daemonset/antrea-agent -n kube-system
+}
+
+setup_flow_visibility() {
+  echo "=== Starting Flow Visibility ==="
+  # install ClickHouse operator
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/clickhouse-operator-install-bundle.yml
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/flow-visibility.yml
+  echo "=== Waiting for ClickHouse and Grafana to be ready ==="
+  sleep 15
+  kubectl wait --for=condition=ready pod -l app=clickhouse-operator -n kube-system --timeout=600s
+  kubectl wait --for=condition=ready pod -l app=clickhouse -n flow-visibility --timeout=600s
+  kubectl wait --for=condition=ready pod -l app=grafana -n flow-visibility --timeout=600s
+}
+
+check_grafana() {
+  echo "=== Check the installation of Grafana plugins ==="
+  grafana_logs=$(kubectl logs -n flow-visibility $(kubectl -n flow-visibility get pod -l app=grafana -o jsonpath="{.items[0].metadata.name}"))
+  if ( echo ${grafana_logs} | grep -q 'msg="Plugin registered" logger=plugin.manager pluginId=antreaflowvisibility-grafana-sankey-plugin' ); then
+    echo "=== antreaflowvisibility-grafana-sankey-plugin installed correctly ==="
+  else
+    echo "=== antreaflowvisibility-grafana-sankey-plugin is NOT installed correctly ==="
+    FAILURE=true
+  fi
+  if ( echo ${grafana_logs} | grep -q 'msg="Plugin registered" logger=plugin.manager pluginId=grafana-clickhouse-datasource' ); then
+    echo "=== grafana-clickhouse-datasource installed correctly ==="
+  else
+    echo "=== grafana-clickhouse-datasource is NOT installed correctly ==="
+    FAILURE=true
+  fi
+}
+
+config_antrea() {
+  echo "=== Stopping Antrea === "
+  kubectl delete -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  echo "=== Enable Antrea Flow Exporter ==="
+  sed -i -e "s/#  FlowExporter: false/  FlowExporter: true/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
```

Comment:
we also need to change the following params in antrea.yaml to make the records to be exported sooner to be compatible with #14 
```
flowCollector:
  flowPollInterval: "1s"
  activeFlowExportTimeout: "2s"
  idleFlowExportTimeout: "1s"
```

Reply:
Ok

Reply:
@wsquan171 I have updated the settings and runs the e2e test code. The clickhouse result seems still have some delays, which caused the related test failed. Any thoughts? 

Here is some output from the test 


test log:
```bash
=== RUN   TestFlowAggregator/IPv4/IntraNodeFlows
    flowaggregator_test.go:913: Network Policies are realized.

    flowaggregator_test.go:870: 
        	Error Trace:	flowaggregator_test.go:870
        	            				flowaggregator_test.go:612
        	            				flowaggregator_test.go:605
        	            				flowaggregator_test.go:209
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestFlowAggregator/IPv4/IntraNodeFlows
        	Messages:   	ClickHouse did not receive the expected records in query output: ; query: SELECT * FROM flows WHERE (sourceIP = '192.168.0.2') AND (destinationIP = '192.168.0.3') AND (sourceTransportPort = 58454)
=== RUN   TestFlowAggregator/IPv4/IntraNodeDenyConnIngressANP
I0507 13:44:51.684314 3040944 framework.go:583] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-reject
I0507 13:44:51.703286 3040944 framework.go:583] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-drop
```

actual result from clickhouse db:

```bash
2022-05-07 13:46:16     2022-05-07 13:45:10     2022-05-07 13:45:23     2022-05-07 13:45:23     2022-05-07 13:45:23     3       192.168.0.2     192.168.0.3     58454   5201    6       356     15091837        356     15091837        249     12956   249       12956   perftest-a      antrea-test     jenkins-theia-e2e-for-pull-request-61-dtnrg     perftest-b      antrea-test     jenkins-theia-e2e-for-pull-request-61-dtnrg     0.0.0.0 0               test-flow-aggregator-networkpolicy-ingress-allow antrea-test              1       1       test-flow-aggregator-networkpolicy-egress-allow antrea-test             1       1       TIME_WAIT       1                       6824272 5824    6824272 6824272 5824    5824    0
2022-05-07 13:46:24     2022-05-07 13:45:10     2022-05-07 13:45:23     2022-05-07 13:45:23     2022-05-07 13:45:23     3       192.168.0.2     192.168.0.3     58454   5201    6       356     15091837        0       0       249     12956   0       0perftest-a       antrea-test     jenkins-theia-e2e-for-pull-request-61-dtnrg     perftest-b      antrea-test     jenkins-theia-e2e-for-pull-request-61-dtnrg     0.0.0.0 0               test-flow-aggregator-networkpolicy-ingress-allow        antrea-test               1       1       test-flow-aggregator-networkpolicy-egress-allow antrea-test             1       1       TIME_WAIT       1                       0       0       0       0       0       0       0
```